osu.Server.Spectator/Hubs/Multiplayer/MultiplayerHub.cs

```diff
@@ -166,6 +164,15 @@ public async Task<MultiplayerRoom> JoinRoomWithPassword(long roomId, string pass
                     }
                 }
 
+                try
+                {
+                    await sharedInterop.AddUserToRoomAsync(Context.GetUserId(), roomId, password);
+                }
+                catch (Exception ex)
+                {
+                    Error("Failed to add user to the databased room", ex);
+                }
```

Comment:
I'm not sure if I misread or just didn't notice when looking at the patch yesterday, but this block executes while the lock on `userUsage` is still held, and I'm not sure how much of a problem to make of it - I'd prefer if this was outside of any locks, but given it's a lock for a specific user only, I could maybe live with it?

Reply:
Makes sense, I've applied this change as well.